### DISCUSSION TITLE
How does the profile import/export code actually work?

### DISCUSSION THREAD

**Discussion by jordanbarnes94** [ebkr/r2modmanPlus#485] [Q&A] [answered]
I find it remarkable that an alphanumeric string 10 characters long can capture all the mod information, including configs, to be shared between users. Can anybody tell me how this is possible? Can the export string really encode everything on its own?

I get that 26^10 is a lot of choice, but what is happening to encode/decode the profile string?

**Comment by ebkr** [comment, marked as answer]
I wish I could do encoding that well :eyes: 
Basically it uploads to a personal hastebin-like instance that uses the end of the URL route as the code.
That part of the URL can then be re-applied and the profile is downloaded.

**Reply by ebkr** [reply]
To add to this, the upload is the export as file encoded as base64.
Downloading will then convert it back to an r2z file where it's then treated like a regular file import.

**Reply by jordanbarnes94** [reply]
You know, this was actually my first instinct. I thought probably the profile is uploaded somewhere and the code is used to retrieve it later, but for some reason I thought it must all be done offline.
